packages/vulcan-routing/lib/modules/splitComponents.jsx

```diff
@@ -48,118 +48,126 @@ const SplitComponentContext = React.createContext("splitComponents");
+  async loadComponent (name) {
+    if (!this.importFunctions[name]) {
+      throw new Error(`Requested load of split-component "${name}" which is not registered`);
+    }
+    if (!this.components[name]) {
+      const componentExports = await this.importFunctions[name]();
+      this.components[name] = componentExports.default;
+    }
+    return this.components[name]
+  },
+
+  // Loads multiple components at the same time
+  async loadComponents(nameArray) {
+    return await Promise.all(nameArray.map(name => this.loadComponent(name)))
+  },
+
+  // Gets a split component by name, if fully loaded. Otherwise returns undefined
+  getComponent(name) {
+    return this.components[name]
+  },
+
+  // Returns Boolean on whether component with given name is registered 
+  isComponentRegistered(name) {
+    return !!this.importFunctions[name]
   }
+}
+
+export function registerSplitComponent(name, importFn) {
+  splitComponentRegistry.registerComponent(name, importFn)
+}
+
+export class SplitComponentCollector {
+  usedComponents = {}
   
-  getComponentsUsedArray() {
-    let ret = [];
-    for(let key in this.componentsUsed)
-      ret.push(key);
-    return ret;
+  getUsedComponents = () => {
+    return Object.keys(this.usedComponents)
   }
   
-  // Either returns the named component, or returns NULL immediately and calls
-  // onReady() later at a time when the component is ready, at which point
-  // calling getComponent again will return the component.
-  getComponent(name, onReady) {
-    if (!this.componentsUsed[name]) {
-      this.componentsUsed[name] = true
-    }
-    
-    if (splitComponentClasses[name]) {
-      return splitComponentClasses[name];
+  getComponent = (name, callback) => {
+    this.usedComponents[name] = true
+    const component = splitComponentRegistry.getComponent(name)
+    if (component) {
+      return component
     } else {
-      loadSplitComponents([name], onReady);
+      splitComponentRegistry.loadComponent(name).then(callback)
+      return null
     }
   }
 }
 
-export class SplitComponentWrapper extends PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      componentsLoaded: 0,
-    }
-  }
-  
-  render() {
-    const { collector, children } = this.props;
-    
-    return (
-      <SplitComponentContext.Provider value={{
-        getComponent: (componentName, onReady) => {
-          return collector.getComponent(componentName, onReady);
-        }
-      }}>
-        {children}
-      </SplitComponentContext.Provider>
-    );
-  }
+
+export const SplitComponentWrapper = ({collector, children}) => {
+  return (
+    <SplitComponentContext.Provider value={{
+      getComponent: collector.getComponent
+    }}>
+      {children}
+    </SplitComponentContext.Provider>
+  );
 }
 
-export class SplitComponent extends PureComponent {
-  componentFinishedLoading = () => {
-    this.forceUpdate();
+
+export class SplitComponentRenderer extends PureComponent {
```

Comment:
This is more of an internal detail of `SplitComponent`, and isn't used outside this file, so make file scoped.

```suggestion
class SplitComponentRenderer extends PureComponent {
```